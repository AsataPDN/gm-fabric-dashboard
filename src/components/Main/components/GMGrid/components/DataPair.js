import styled from "styled-components";
import { PropTypes } from "prop-types";

import { COLOR_CONTENT_BACKGROUND } from "style/styleVariables";
import { spacingScale, edgeColor } from "style/styleFunctions";

const DataPair = styled.div`
  padding-right: 15%;
  padding-top: ${spacingScale(0.25)};
  padding-bottom: ${spacingScale(0.25)};
  display: flex;
  flex-direction: row;
  position: relative;
  ${props => (props.priority === "primary" ? dataPairPrimary() : "")};
  ${props =>
    props.priority === "normal" || props.priority === "secondary"
      ? "font-size: 0.8em;"
      : ""};
`;

DataPair.propTypes = {
  priority: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
};

const dataPairPrimary = () => {
  return `
    overflow: hidden;
    &:after {
      content: "";
      position: absolute;
      bottom: 0;
      left: 4em;
      right: 4em;
      border-top: 1px solid ${edgeColor(COLOR_CONTENT_BACKGROUND)};
      border-left: 8em solid transparent;
      border-right: 8em solid transparent;
    }
`;
};

export default DataPair;
