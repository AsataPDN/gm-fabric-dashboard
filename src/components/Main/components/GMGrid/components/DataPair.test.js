import React from "react";
import { shallow } from "enzyme";
import DataPair from "./DataPair";

describe("DataPair", () => {
  it("matches snapshot when priority is primary", () => {
    const aDataPair = shallow(<DataPair priority="primary" />);
    expect(aDataPair).toMatchSnapshot();
  });
  it("matches snapshot when priority is normal", () => {
    const aDataPair = shallow(<DataPair priority="normal" />);
    expect(aDataPair).toMatchSnapshot();
  });
});
