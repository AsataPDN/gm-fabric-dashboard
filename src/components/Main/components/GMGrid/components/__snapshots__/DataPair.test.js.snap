// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataPair matches snapshot when priority is normal 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 0.8em;
  padding-right: 15%;
  padding-top: 2px;
  padding-bottom: 2px;
  position: relative;
}

<div
  className="c0"
/>
`;

exports[`DataPair matches snapshot when priority is primary 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-right: 15%;
  padding-top: 2px;
  padding-bottom: 2px;
  position: relative;
  overflow: hidden;
}

.c0:after {
  content: "";
  position: absolute;
  bottom: 0;
  left: 4em;
  right: 4em;
  border-top: 1px solid hsl(0,0%,92%);
  border-left: 8em solid transparent;
  border-right: 8em solid transparent;
}

<div
  className="c0"
/>
`;
